reVX/pras/rev_reeds_pras.py

```diff
@@ -0,0 +1,736 @@
+        plexos_id to each zone so that this can be used as a plexos node table
+        for calling super().__init__()
+
+        Parameters
+        ----------
+        reeds_build : str | pd.DataFrame
+            ReEDS buildout with rows for built capacity (MW) at each reV SC
+            point. This should have columns: reeds_year, built_capacity, and
+            sc_gid (corresponding to the reV supply curve point gid). Some
+            cleaning of the column names will be performed for legacy tables
+            but these are the column headers that are desired.
+
+        Returns
+        -------
+        pras_build_zones : pd.DataFrame
+            pras zones DataFrame with all available zones in reeds build table
+            and corresponding plexos_id
+        """
+        pras_build_zones = reeds_build
+        if isinstance(pras_build_zones, str):
+            pras_build_zones = pd.read_csv(pras_build_zones)
+        class_mask = pras_build_zones['class'] == self._res_class
+        pras_build_zones = pras_build_zones[class_mask]
+        year_mask = pras_build_zones['year'] == self._build_year
+        pras_build_zones = pras_build_zones[year_mask]
+        pras_build_zones = pras_build_zones[['latitude', 'longitude',
+                                             'region', 'built_capacity']]
+        pras_build_zones_grp = pras_build_zones.groupby(by='region')
+        pras_build_zones = pras_build_zones_grp.mean()
+        cap = pras_build_zones_grp.sum()['built_capacity']
+        pras_build_zones['built_capacity'] = cap
+        pras_build_zones = pras_build_zones.reset_index()
+        pras_build_zones['plexos_id'] = np.arange(len(pras_build_zones))
+        return pras_build_zones
+
+    def _init_output_files(self, pras_file):
+        """
+        Initialize the output file to be written with new generation profiles
+        and define the qa file name.
+
+        Parameters
+        ----------
+        pras_file : str
+            Path to original pras file containing generation profiles to be
+            updated. The file will be copied to a new path (given by
+            _output_file) and then updated with the new profiles
+        """
+        if not self._pras_file.endswith('.h5'):
+            self._output_file += '.h5'
+        self._output_file = pras_file.replace('.h5', '_updated.h5')
+        msg = ("pras_file={} does not exist. This file must come from a "
+               "previous ReEDS2PRAS run").format(self._pras_file)
+        assert os.path.exists(self._pras_file), msg
+
+        if not os.path.exists(self._output_file):
+            shutil.copy(self._pras_file, self._output_file)
+
+        self._qa_file = self._output_file.replace(
+            '.h5', '_qa_{}_{}.csv'.format(self._tech_type, self._res_class))
+
+        self._prof_file = self._output_file.replace(
+            '.h5', '_{}_{}.csv'.format(self._tech_type, self._res_class))
+
+    @classmethod
+    def get_pras_meta(cls, pras_file):
```

Comment:
I would probably make this a property with a self._pras_meta, but fine as is. 

Actually, wait, this is a thing further down, whats the diff?